packages/react/src/components/HotspotEditorModal/HotspotEditorDataSourceTab/HotspotEditorDataSourceTab.test.jsx

```diff
@@ -146,19 +163,44 @@ describe('HotspotEditorDataSourceTab', () => {
     });
   });
 
-  it('pops the data items modal', () => {
+  it('pops the data items modal with downsample methods', async () => {
     const onChange = jest.fn();
+    const onEditDataItem = jest.fn().mockImplementation(() => [
+      { id: 'none', text: 'None' },
+      { id: 'max', text: 'Maximum' },
+      { id: 'min', text: 'Minimum' },
+    ]);
     render(
       <HotspotEditorDataSourceTab
         hotspot={cardConfigWithPresets.content.hotspots[0]}
         cardConfig={cardConfigWithPresets}
-        dataItems={dataItems}
+        dataItems={dataItemsV2}
         onChange={onChange}
         translateWithId={() => {}}
+        onEditDataItem={onEditDataItem}
       />
     );
     userEvent.click(screen.getAllByRole('button')[2]);
-    // Card config with the elevators hotspot removed
+    await waitFor(() => expect(screen.getByRole('dialog')).toBeInTheDocument());
+
+    expect(screen.getByRole('dialog')).toBeInTheDocument();
+  });
+  it('pops the data items modal without downsample methods', async () => {
+    const onChange = jest.fn();
+    const onEditDataItem = jest.fn().mockImplementation(() => []);
+    render(
+      <HotspotEditorDataSourceTab
+        hotspot={cardConfigWithPresets.content.hotspots[0]}
+        cardConfig={cardConfigWithPresets}
+        dataItems={dataItemsV2}
+        onChange={onChange}
+        translateWithId={() => {}}
+        onEditDataItem={onEditDataItem}
+      />
+    );
+    userEvent.click(screen.getAllByRole('button')[2]);
+    await waitFor(() => expect(screen.getByRole('dialog')).toBeInTheDocument());
```

Comment:
Isn't this the same assertion that we have on 204?

Reply:
Yes I removed 204 and 186